modules/conversation/router.js

```diff
@@ -0,0 +1,92 @@
+const debug = require('debug')('server:conversation:router');
+const router = require('express').Router();
+const conversationService = require('./service');
+const conversationMiddleware = require('./middleware');
+const _ = require('underscore');
+
+router.post('/conversations', (req, res) => {
+    debug('POST /conversations');
+    debug('Body Content: ' + JSON.stringify(req.body, null, 2));
+
+    conversationService.createConversation(req.body.title, req.body.access, req.session.user._id)
+        .then((conversation) => {
+            res.status(200).json(conversation);
+        })
+        .catch((err) => {
+            console.error(err);
+            res.status(500).send(err);
+        });
+});
+
+router.put('/conversations/:id', (req, res) => {
+    debug('PUT /conversations/' + req.params.id);
+    debug('Body Content: ' + JSON.stringify(req.body, null, 2));
+
+    const properties = _.pick(req.body, 'title', 'access', 'moderators', 'members');
+
+    conversationService.updateConversation(req.params.id, properties)
+        .then((conversation) => {
+            res.status(200).json(conversation);
+        })
+        .catch((err) => {
+            console.error(err);
+            res.status(500).send(err);
+        });
+});
+
+router.get('/conversations/:id', (req, res) => {
+    debug('GET /conversations/' + req.params.id);
+
+    conversationService.getConversation(req.params.id, req.session.user._id)
+        .then((conversation) => {
+            res.status(200).json(conversation);
+        })
+        .catch((err) => {
+            console.error(err);
+            res.status(500).send(err);
+        });
+});
+
+router.get('/conversations', (req, res) => {
+    debug('GET /conversations?filter=' + req.query.filter);
+
+    let getConversationStrategy = conversationService.getUserConversations;
+    let isUserRelatedFunctionCall = true;
+
+    if (req.query.filter === 'public') {
+        isUserRelatedFunctionCall = false;
+        getConversationStrategy = conversationService.getPublicConversations;
+    }
+
+    let promise;
+
+    if (isUserRelatedFunctionCall) {
```

Comment:
Die Variablen isUserRelatedFunctionCall und getConversationStrategy scheint mir hier etwas überflüssig. Würde es nicht reichen in diese If-Abfrage einfach req.query.filter === 'public' zu packen und die entsprechende conversationStrategy function direct mit dem richtigen Aufruf dem Promise zuzuweisen?